eval/src/vespa/eval/eval/value_codec.cpp

```diff
@@ -308,13 +310,19 @@ void encode_value(const Value &value, nbostream &output) {
 }
 
 std::unique_ptr<Value> decode_value(nbostream &input, const ValueBuilderFactory &factory) {
-    Format format(input.getInt1_4Bytes());
-    ValueType type = decode_type(input, format);
-    size_t num_mapped_dims = type.count_mapped_dimensions();
-    size_t dense_subspace_size = type.dense_subspace_size();
-    const size_t num_blocks = maybe_decode_num_blocks(input, (num_mapped_dims > 0), format);
-    DecodeState state{type, dense_subspace_size, num_blocks, num_mapped_dims};
-    return typify_invoke<1,TypifyCellType,ContentDecoder>(type.cell_type(), input, state, factory);
+    try {
+        Format format(input.getInt1_4Bytes());
+        ValueType type = decode_type(input, format);
+        size_t num_mapped_dims = type.count_mapped_dimensions();
+        size_t dense_subspace_size = type.dense_subspace_size();
+        const size_t num_blocks = maybe_decode_num_blocks(input, (num_mapped_dims > 0), format);
+        DecodeState state{type, dense_subspace_size, num_blocks, num_mapped_dims};
+        return typify_invoke<1,TypifyCellType,ContentDecoder>(type.cell_type(), input, state, factory);
+    } catch (const OOMException &) {
+        throw;
+    } catch (const Exception &e) {
+        throw DecodeValueException("failed to decode value", e);
```

Comment:
this means we need to add catch of DecodeValueException in those places where the old exception was handled.  at least this one:
```
document/src/vespa/document/fieldvalue/document.cpp:273:    } catch (const IllegalStateException &e) {
```
need to also review these:
```
searchcore/src/apps/proton/proton.cpp:246:    } catch (const vespalib::IllegalStateException & e) {
searchcore/src/vespa/searchcore/proton/server/documentdb.cpp:616:    } catch (const vespalib::IllegalStateException & e) {
searchcore/src/vespa/searchcore/proton/server/feedhandler.cpp:356:    } catch (const IllegalStateException & e) {
searchcorespi/src/vespa/searchcorespi/index/indexmaintainer.cpp:1174:    } catch (const vespalib::IllegalStateException & e) {
searchlib/src/vespa/searchlib/docstore/filechunk.cpp:189:                } catch (const vespalib::IllegalStateException & e) {
storageserver/src/apps/storaged/storage.cpp:175:    } catch (const vespalib::IllegalStateException & e) {
```

Reply:
The DecodeValueException is now converted the the already used DeserializationException in document context. Also, catching the IllegalState exception was not completely useful, since the exception we got was IllegalArgument.